src/main/java/com/splunk/kafka/connect/SplunkSinkConnectorConfig.java

```diff
@@ -401,13 +401,30 @@ private String getMetaForTopic(String[] metas, int expectedLength, int curIdx, S
     }
 
     private Map<String, Map<String, String>> initMetaMap(Map<String, String> taskConfig) {
-        String[] topics = split(taskConfig.get(SinkConnector.TOPICS_CONFIG), ",");
+        /* 
+        ** Modified 2021-08-28 by R. Vasaly to allow the use of 'topics.regex' instead
+        ** of the static list of topics. If topics.regex is specified in the config,
+        ** the Connector will subscribe to all matching topics. If topics.regex is used,
+        ** a single dummy entry is stored in the 'topics' string array so that existing
+        ** code does not fail with a null string. With the single dummy topic, mapping from
+        ** topic value to Splunk metadata will not work, so either the Headers must define
+        ** the Splunk metadata, or simply rely on the HEC token to set default index,
+        ** sourcetype, etc.
+        */
+        String[] topics = new String[1];
+        String[] topics_conf = split(taskConfig.get(SinkConnector.TOPICS_CONFIG), ",");
         String[] topicIndexes = split(indexes, ",");
         String[] topicSourcetypes = split(sourcetypes, ",");
         String[] topicSources = split(sources, ",");
 
         Map<String, Map<String, String>> metaMap = new HashMap<>();
         int idx = 0;
+        // If the config has no "topics" values, set the array to a dummy entry
+        if (topics_conf == null || topics_conf.length == 0 || topics_conf[0].length() == 0) {
```

Comment:
Hi @rnvasaly,
I don’t think we need any dummy topic here, and this metaMap is not useful for topics.regex functionality. 
I face below two issues

1. For a topic.regex support, Splunk metadata fields("splunk.indexes", "splunk.sourcetypes", "splunk.sources"), parameters are ignored and should be omitted. but if anyone passes multiple values for these fields then got ConfigException. this is not the right behavior. So instead of this we simply wrap for loop with this condition 
`if(topics != null && topics.length != 0)`,  then it will not raise any unnecessary error and simply omitted splunk metadata configuration.

2. For defining Splunk metadata on a per-event basis by using Kafka Header Fields ("splunk.header.index", "splunk.header.sourcetype", etc.). we need to add a null check in headerId() method of SplunkSinkTask class like given below.
`if(indexHeader != null) {
            headerString.append(indexHeader.value().toString());
        } else {
            if(metas!=null)
            headerString.append(metas.get("index"));
        }`
        which overcome nullPointerExecption when any kafka header field missing in records and also handle code flow break issue.

Let me know if need more clarification or see any issue.